Automattic/woocommerce-payments
# Render ECE buttons

Closes #8772 
Closes #8825 

#### Changes proposed in this Pull Request

Render the ECE buttons behind a feature flag. The buttons are rendered in:

- The product page.
- The shortcode cart/checkout.
- The blocks cart/checkout.

It also relocates functions that were used by other parts of the application, such as the WooPay button, from `payment-request/utils` to a new high-level utility directory, `utils/express-checkout`. This move aims to decouple these helpers from any specific express checkout implementation.

#### Testing instructions

Before you enable the feature flag, [test the PRB buttons](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#checkout-with-payment-request) and ensure there are no regressions. Also [test the WooPay express checkout button](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#woopay-checkout).

- Enable the ECE feature flag. `npm run wp option update _wcpay_feature_stripe_ece 1`
- Confirm that the ECE buttons are rendering on the shortcode and blocks product/cart/checkout pages.
- Ensure that the PRB buttons are not rendered when the feature flag is enabled.
- Ensure the ECE buttons are not rendered when the feature flag is disabled.

-------------------


**Post merge**
